picard/ui/profileeditor.py

```diff
@@ -0,0 +1,456 @@
+    EXPANDED_KEY = "profile_settings_tree_expanded_list"
+
+    TREEWIDGETITEM_COLUMN = 0
+
+    options = [
+        IntOption("persist", POSITION_KEY, 0),
+        ListOption("persist", EXPANDED_KEY, [])
+    ]
+
+    signal_save = QtCore.pyqtSignal()
+
+    def __init__(self, parent=None):
+        """User profile editor.
+        """
+        super().__init__(parent)
+
+        self.main_window = parent
+
+        self.setWindowTitle(_(self.TITLE))
+        self.displaying = False
+        self.loading = True
+        self.ui = Ui_ProfileEditorPage()
+        self.ui.setupUi(self)
+        self.setModal(True)
+
+        self.make_buttons()
+
+        self.ui.profile_editor_splitter.setStretchFactor(1, 1)
+        self.move_view = MoveableListView(self.ui.profile_list, self.ui.move_up_button,
+                                          self.ui.move_down_button)
+
+        self.ui.profile_list.currentItemChanged.connect(self.current_item_changed)
+        self.ui.profile_list.itemSelectionChanged.connect(self.item_selection_changed)
+        self.ui.settings_tree.itemChanged.connect(self.update_profile_changed_message)
+        self.ui.settings_tree.expanded.connect(self.update_current_expanded_items_list)
+
+        self.current_profile_id = None
+        self.profiles_changed = False
+        self.expanded_sections = []
+        self.building_tree = False
+
+        self.load()
+        self.loading = False
+
+    def make_buttons(self):
+        """Make buttons and add them to the button bar.
+        """
+        self.ui.make_it_so.setToolTip(_("Save all profile information to the user settings"))
+        self.ui.make_it_so.clicked.connect(self.make_it_so)
+
+        self.new_profile_button = QtWidgets.QPushButton(_('New'))
+        self.new_profile_button.setToolTip(_("Create a new profile"))
+        self.new_profile_button.clicked.connect(self.new_profile)
+        self.ui.buttonbox.addButton(self.new_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
+
+        self.copy_profile_button = QtWidgets.QPushButton(_('Copy'))
+        self.copy_profile_button.setToolTip(_("Copy to a new profile"))
+        self.copy_profile_button.clicked.connect(self.copy_profile)
+        self.ui.buttonbox.addButton(self.copy_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
+
+        self.delete_profile_button = QtWidgets.QPushButton(_('Delete'))
+        self.delete_profile_button.setToolTip(_("Delete the profile"))
+        self.delete_profile_button.clicked.connect(self.delete_profile)
+        self.ui.buttonbox.addButton(self.delete_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
+
+        self.save_profile_button = QtWidgets.QPushButton(_('Save'))
+        self.save_profile_button.setToolTip(_("Save changes to the profile"))
+        self.save_profile_button.clicked.connect(self.save_profile)
+        self.ui.buttonbox.addButton(self.save_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
```

Comment:
This should be `ApplyRole`

Reply:
I actually removed it completely to simplify the UI.  Changes are now saved locally (within the dialog) as they are made.